### ISSUE TITLE
Intermittent 502 in API polls causes UI to redirect

### ISSUE THREAD

**Issue by colearendt** [kubernetes/dashboard#5056]
##### Environment
```
Installation method: kubernetes YAML file
Kubernetes version:
Dashboard version: 2.0.0-rc7
Operating system:
Node.js version ('node --version' output):
Go version ('go version' output):
```

##### Steps to reproduce
Put a proxy in front of the Kubernetes dashboard that intermittently responds  (i.e. 1 out of every 10-20k) with a 502 to the polls for things like: `https://kubernetes.staging.soleng.rstudioservices.com/kubernetes/api/v1/configmap/default?itemsPerPage=10&page=1&sortBy=d,creationTimestamp`

##### Observed result

The dashboard redirects to `#/error?namespace=default` and presents an ugly UI:

![image](https://user-images.githubusercontent.com/23075542/78814533-c7abb700-799c-11ea-9bb9-d305982cde46.png)

##### Expected result
The dashboard UI should try again with some backoff interval, and should not surface the bad response of this polling to end users.

##### Comments
This is mostly a problem with my proxy, but wanted to mention it because it feels like an instability in the kubernetes dashboard as well. Handling networking blips more gracefully would definitely give the user a better experience!

**Comment by maciaszczykm** [comment]
True, this should be improved. However, priority is rather low right now. Thanks for reporting.

**Comment by 4ng31** [comment]
Hi, I have the exact same behavior in my setup. I have a k8s deployment with an HAproxy LoadBalancer in front of the API servers and I'm having an intermittent 502 at the Kubernetes Dashboard. Even if use a keepalive connection for about 60 seconds.

`kubectl proxy --accept-hosts='.*' --keepalive=60s`
